libs/platform/src/lib/approval-flow/approval-flow-user-list/approval-flow-user-list.component.ts

```diff
@@ -122,15 +130,28 @@ export class ApprovalFlowUserListComponent implements AfterViewInit, OnChanges,
 
     /** @hidden */
     _onSelect(event: SelectionChangeEvent): void {
-        this._selectedItems = event.selectedItems;
+        if (!this._dontCallSelection) {
+            this._selectedItems = event.selectedItems;
+
+            if (event.removed) {
+                this._allSelectedUsers = this._allSelectedUsers.filter(
+                    (user) => `${this._idPrefix + user.id}` !== event.removed.id
+                );
+            }
+
+            if (event.added) {
+                this._allSelectedUsers.push(...this._getUsersFromSelectedItems([event.added]));
+            }
 
-        this.onSelectionChange.emit(this._getUsersFromSelectedItems(event.selectedItems));
+            this.onSelectionChange.emit(this._allSelectedUsers);
+            this._cdr.detectChanges();
+        }
     }
 
     /** @hidden */
     private _getUsersFromSelectedItems(items: BaseListItem[]): ApprovalUser[] {
         return items
-            .map((item) => this.users.find((user) => `${this._idPrefix + user.id}` === item.itemEl.nativeElement.id))
+            .map((item) => this.users.find((user) => `${this._idPrefix + user.id}` === item.id))
             .filter((u): u is ApprovalUser => !!u);
```

Comment:
Change it due to DOM query on potential large user display on dialog which can cause bottleneck on rendering.